src/Simplex_tree/include/gudhi/Simplex_tree/multi_filtrations/Finitely_critical_filtrations.h

```diff
@@ -0,0 +1,157 @@
+    int n = std::min(a.size(), b.size());
+    for (int i = 0; i < n; ++i) {
+      if (a[i] > b[i]) return false;
+    }
+    return true;
+  }
+
+  // GREATER THAN OPERATORS
+  friend bool operator>(const Finitely_critical_multi_filtration& a, const Finitely_critical_multi_filtration& b) {
+    return b < a;
+  }
+  friend bool operator>=(const Finitely_critical_multi_filtration& a, const Finitely_critical_multi_filtration& b) {
+    return b <= a;
+  }
+
+  Finitely_critical_multi_filtration& operator=(const Finitely_critical_multi_filtration& a) {
+    std::vector<T>::operator=(a);
+    return *this;
+  }
+
+  std::vector<T>& _convert_back() { return *this; }
+
+  friend Finitely_critical_multi_filtration& operator-=(Finitely_critical_multi_filtration& result,
+                                                        const Finitely_critical_multi_filtration& to_substract) {
+    std::transform(result.begin(), result.end(), to_substract.begin(), result.begin(), std::minus<T>());
+    return result;
+  }
+  friend Finitely_critical_multi_filtration<T>& operator+=(Finitely_critical_multi_filtration<T>& result,
+                                                           const Finitely_critical_multi_filtration& to_add) {
+    std::transform(result.begin(), result.end(), to_add.begin(), result.begin(), std::plus<T>());
+    return result;
+  }
+
+  friend Finitely_critical_multi_filtration& operator-=(Finitely_critical_multi_filtration& result,
+                                                        const T& to_substract) {
+    // std::transform(result.begin(), result.end(), to_substract.begin(),result.begin(), std::minus<T>());
+    for (auto& truc : result) {
+      truc -= to_substract;
+    }
+    return result;
+  }
+  friend Finitely_critical_multi_filtration<T>& operator+=(Finitely_critical_multi_filtration<T>& result,
+                                                           const T& to_add) {
+    for (auto& truc : result) {
+      truc += to_add;
+    }
+    return result;
+  }
+
+  // template<class array_like>
+  friend bool operator==(Finitely_critical_multi_filtration<T>& self,
+                         const Finitely_critical_multi_filtration<T>& to_compare) {
+    if (self.size() != to_compare.size()) return false;
+    auto it = to_compare.begin();
+    for (auto i = 0u; i < self.size(); i++) {
+      if (self.at(i) != *(it++)) return false;
+    }
+    return true;
+  }
+
+  static std::vector<std::vector<T>> to_python(const std::vector<Finitely_critical_multi_filtration<T>>& to_convert) {
+    return std::vector<std::vector<T>>(to_convert.begin(), to_convert.end());
+  }
+
+  static std::vector<Finitely_critical_multi_filtration<T>> from_python(const std::vector<std::vector<T>>& to_convert) {
+    return std::vector<Finitely_critical_multi_filtration<T>>(to_convert.begin(), to_convert.end());
+    ;
+  }
+  void push_to(const Finitely_critical_multi_filtration<T>& x) {
```

Comment:
Once you do the description, you could add that this method is mandatory for `Filtration_value` in the simplex tree.

Reply:
I've added that to the concept